Python/executor_cases.c.h

```diff
@@ -118,12 +118,38 @@
             break;
         }
 
+        case TO_BOOL: {
+            static_assert(INLINE_CACHE_ENTRIES_TO_BOOL == 3, "incorrect cache size");
```

Comment:
That seems a bit harsh. It would mean that if we have a sequence of innocent bytecodes like
```
LOAD_FAST i
LOAD_CONST j
BINARY_OP (/)
STORE_FAST k
...
```
we would not be able to include the `BINARY_OP` and `STORE_FAST` instructions in a superblock, because the unspecialized `BINARY_OP` ends the superblock generation.

Reply:
Harsh, but fair. `BINARY_OP` needs to work harder at being specialized 🙂 

Seriously though, there is no merit in longer superblocks if we can't optimize them well, and the unspecialized forms do not optimize well. They are much slower, and force us to discard a lot of type information.

It is far more profitable to improve the specializer, so that unspecialized instructions are rare.